osu.Server.Spectator/Hubs/Metadata/MetadataHub.cs

```diff
@@ -180,7 +180,7 @@ public async Task<MultiplayerPlaylistItemStats[]> BeginWatchingMultiplayerRoom(l
 
             using var db = databaseFactory.GetInstance();
 
-            MultiplayerRoomStats stats = (await cache.GetOrCreateAsync<MultiplayerRoomStats>(id.ToString(), e =>
+            MultiplayerRoomStats stats = (await cache.GetOrCreateAsync<MultiplayerRoomStats>($@"{nameof(MultiplayerRoomStats)}#{id}", e =>
```

Comment:
Reason for this weird change is that I wanted to use a memory cache for the build lookups to reduce redundancy and https://github.com/ppy/osu-server-spectator/blob/bab7c8c8db1803bbd19decb418dae1672b6986b6/osu.Server.Spectator/Startup.cs#L60 registers the cache instance as a singleton:

![Image](https://github.com/user-attachments/assets/7fb44fb8-5240-4036-bec1-f244fd46e4ea)

so cache keys have to be globally unique lest some Really Dumb Stuff Happens™